### PR TITLE
feature: resolving snap desktop files using app armor as the primary source

### PR DESCRIPTION
**Pull request by mattkae**
## What's new?
- AppArmor is now the primary source of snap desktop files in the `DesktopFileManager`

### CODE REVIEW

**Comment by mattkae** [comment]
Whoops, broke the tests. Will fix em soon...

**Review by Saviq — CHANGES_REQUESTED** [review]
You'll need to add a `libapparmor` build dependency to `debian/control`, and to `snap/snapcraft.yaml`.

File: src/server/frontend_wayland/desktop_file_manager.cpp [hunk 1]
```diff
@@ -157,8 +159,27 @@ std::string mf::DesktopFileManager::parse_snap_security_profile_to_desktop_id(st
     return sandboxed_app_id + DESKTOP_FILE_POSTFIX;
 }
 
-std::shared_ptr<mf::DesktopFile> mf::DesktopFileManager::resolve_if_snap(int pid)
+std::shared_ptr<mf::DesktopFile> mf::DesktopFileManager::resolve_if_snap(int pid, mir::Fd const& socket_fd)
 {
+    // First, try to resolve the app armor profile
```
Comment (Saviq) [inline src/server/frontend_wayland/desktop_file_manager.cpp:164]:
```suggestion
    // First, try to resolve the AppArmor profile
```

**Comment by mattkae** [comment]
@Saviq What should I do about the apparmor dependency on Fedora and Alpine? :thinking: Should that be an optional dependency?

**Comment by Saviq** [comment]
> @Saviq What should I do about the apparmor dependency on Fedora and Alpine? :thinking: Should that be an optional dependency?

If they're not there, yeah.

**Comment by Saviq** [comment]
@mattkae Alpine does have it, BTW:

https://pkgs.alpinelinux.org/packages?name=apparmor

**Comment by mattkae** [comment]
> @mattkae Alpine does have it, BTW:
> 
> https://pkgs.alpinelinux.org/packages?name=apparmor

Do you know if I'll need something else there too? https://github.com/canonical/mir/actions/runs/9552401270/job/26328915493?pr=3426

**Comment by Saviq** [comment]
> Do you know if I'll need something else there too? https://github.com/canonical/mir/actions/runs/9552401270/job/26328915493?pr=3426

Yeah, you need to install it! :D

https://github.com/canonical/mir/blob/31ef0787e794f23e1eca415f575cd5859ec096b4/spread/build/alpine/task.yaml#L6-L34

**Comment by mattkae** [comment]
> mir/spread/build/alpine/task.yaml

I have indeed added it, to no avail: https://github.com/canonical/mir/actions/runs/9566199163/job/26370892700?pr=3426 :cry: Is there _something else_ that I'm missing here?

**Comment by mattkae** [comment]
> _lib_apparmor

That is what I have ;) https://github.com/canonical/mir/pull/3426/commits/918237ea740566a29b937d6a880fa8d246f1c6e8#diff-5f6d75408d28a1d92efc5968e0b1a34ee3c512d14427c514e56cd0b23290f52d

**Review by RAOF — COMMENTED** [review]
Looks sensible, but the build system parts could be cleaned up.

File: doc/sphinx/.readthedocs.yaml [hunk 2]
```diff
@@ -35,6 +35,7 @@ build:
   - libxml++2.6-dev
   - libyaml-cpp-dev
   - xsltproc
+  - libapparmor-dev
```
Comment (Saviq) [inline doc/sphinx/.readthedocs.yaml:38]: Since it's optional, let's skip it here.

**Review by Saviq — APPROVED** [review]
Looks legit to me.

**Review by Saviq — APPROVED** [review]
Yeah!

**Pull request merged**